cf_units/__init__.py

```diff
@@ -602,10 +524,11 @@ def _num2date_to_nearest_second(time_value, utime):
     ceil_mask = np.logical_or(has_half_seconds, microseconds >= 500000)
     if time_values[ceil_mask].size > 0:
         useconds = Unit('second')
-        second_frac = useconds.convert(0.75, utime.units)
-        dates[ceil_mask] = utime.num2date(time_values[ceil_mask] + second_frac)
+        second_frac = useconds.convert(0.75, time_units)
+        dates[ceil_mask] = cftime.num2date(
+            time_values[ceil_mask] + second_frac, **num2date_kwargs)
     dates[round_mask] = _discard_microsecond(dates[round_mask])
-    result = dates[0] if shape is () else dates.reshape(shape)
+    result = dates[0] if shape == () else dates.reshape(shape)
```

Comment:
These were giving me SyntaxWarnings (python3.9) so I thought I might as well change them while I'm in there.